pandas/tests/indexes/test_numeric.py

```diff
@@ -944,6 +944,11 @@ def test_constructor(self):
         res = Index(np.array([-1, 2 ** 63], dtype=object))
         tm.assert_index_equal(res, idx)
 
+        # https://github.com/pandas-dev/pandas/issues/29526
+        idx = UInt64Index([1, 2 ** 63 + 1], dtype=np.uint64)
+        res = Index([1, 2 ** 63 + 1], dtype=np.uint64)
```

Comment:
Maybe a little more direct to have this be `UInt64Index` initialized from an array that doesn't need to go through the conversion, i.e. `UInt64Index(np.array([1, 2**63 + 1, dtype=np.uint64]))`